source/plugins/go/src/oms.go

```diff
@@ -1622,6 +1699,25 @@ func GetContainerIDK8sNamespacePodNameFromFileName(filename string) (string, str
 	return id, ns, podName, containerName
 }
 
+// GetDSNameAndDeploymentNameFromK8sPodName tries to extract potential deployment name and daemonset name from the pod name. If its not possible to extract then sends empty string
+func GetDSNameAndDeploymentNameFromK8sPodName(podName string) (string, string) {
```

Comment:
This should be GetControllerNameFromK8sPodName and also this should cover other controller types jobs, cronjobs and statefulset etc.